### PR TITLE
Add test for exactmatch

### PR DESCRIPTION
**Pull request by justinemarie**
Panda wins the lottery.

### CODE REVIEW

File: bessctl/conf/testing/module_tests/exact_match.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# Crash test -- generate a bunch of rules and stuff packets through
+em0 = ExactMatch(fields=[{'offset': 23, 'size': 1},  # random fields, I have no idea what these are
+                         {'offset': 2, 'size': 2},
+                         {'offset': 29, 'size': 1}])
+
+em0.add(fields=[chr(255), chr(0) + chr(12), chr(55)], gate=0)
```
Comment (sangjinhan) [inline bessctl/conf/testing/module_tests/exact_match.py]: Not a big deal, but `[b'\xff', b'\x00\x0c\x37']` would be more representative. After all, `bytes` should be preferred than `str`.

File: bessctl/conf/testing/module_tests/exact_match.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+em0.add(fields=[chr(255), chr(12) + chr(34), chr(255)], gate=2)
+em0.add(fields=[chr(19), chr(12) + chr(34), chr(255)], gate=3)
+em0.add(fields=[chr(19), chr(66) + chr(0), chr(2)], gate=4)
+em0.add(fields=[chr(123), chr(4) + chr(0), chr(2)], gate=5)
+
+CRASH_TEST_INPUTS.append([em0, 1, 6])
+
+
+#Output test -- just make sure packets go out right ports
+em1 = ExactMatch(fields=[{'offset': 26, 'size': 4},
```
Comment (sangjinhan) [inline bessctl/conf/testing/module_tests/exact_match.py]: Can we have some metadata-based fields for better coverage?

File: bessctl/conf/testing/module_tests/exact_match.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+OUTPUT_TEST_INPUTS.append([em1, 1, 4,
+                           [{'input_port': 0,
+                             'input_packet': test_packet1,
+                             'output_port': 1,
+                             'output_packet': test_packet1},
+                            {'input_port': 0,
+                             'input_packet': test_packet2,
+                             'output_port': 2,
+                             'output_packet': test_packet2},
+                               {'input_port': 0,
```
Comment (changlan) [inline bessctl/conf/testing/module_tests/exact_match.py]: Indentation here is strange. Was `autopep8` not working..?

**Comment by justinemarie** [comment]
The autopep indentation was harder to follow -- it had everything over so
far, I manually pulled it in to the left.

**Comment by justinemarie** [comment]
(I can go with the normal autopep output if that's what we prefer)

File: bessctl/conf/testing/module_tests/exact_match.py (hunk 3 above)
Comment (changlan) [inline bessctl/conf/testing/module_tests/exact_match.py]: Not a big deal. Just noticed these entries are not aligned.

**Comment by justinemarie** [comment]
Will do!

**Comment by changlan** [comment]
image

**Pull request merged**